### ISSUE TITLE
Create a Codegen REST API

### ISSUE THREAD

**Issue by manekinekko** [xlayers/xlayers#191]
**Is your feature request related to a problem? Please describe.**
xLayers provides a nice and easy UI for users. But this UI isn't suited for batch codegen nor for other automated tasks.

**Describe the solution you'd like**
We want to provide a REST API that will allow developers to send in a SketchApp design and get the generated code. The API should support batch codegen.

**Additional context**
When this gets approved, it should be implemented in a separate repository.

**Comment by manekinekko** [comment]
We are use @nestjs framework for the REST API.

**Comment by MrCube42** [comment]
@manekinekko Looks interesting to me, I could help (built some REST APIs using Nest). Sure you want this to be in an own repository rather than mono repo? Had good experiences with 2 directory like */frontend* and */backend* where each has its own `package.json` in a single repository.

**Comment by manekinekko** [comment]
Thank you @MrCube42 for this. We were waiting until we publish all different packages to NPM. Eventually, we'd like for this to be in a separate repository.

In the meantime, do you wanna draft a design doc for this before starting the implementation? Please find the design doc template in the [wiki section](https://github.com/xlayers/xlayers/wiki/Design-doc-library).

**Comment by MrCube42** [comment]
@manekinekko Sure. I'll have a look and let you know if I had a first draft ready. May take some time though. 🕰

**Comment by MrCube42** [comment]
@manekinekko I created a design document draft https://docs.google.com/document/d/1PN7eQ8d12JgRusO-1pmVuwJXm2HhsLtp4xRGtwfvxvU/edit?usp=sharing. It is in a very early state. Especially the implementation detail needs more work. I tried to fill in everything I can as a starting point and added some initial ideas. I made it available for commenting.

**Comment by manekinekko** [comment]
Hey @MrCube42 sorry about this delay. 

Your design doc lgtm. Do you still have time to work on a POC? I've created this repo (https://github.com/xlayers/xlayers-backend-nestjs) where we can push your code once you're ready.

Cheers.

**Comment by MrCube42** [comment]
Hey @manekinekko. Thanks for the feedback and reaching out to me again. It's been a while. I completely forgot about this on my personal backlog 😁 I am very unsure about time and motivation at the moment. However I'll put it into my latest bookmarks and will have a look at it again. To be honest, it has low priority for me at the moment, so anyone should feel free build on top of my very first design doc steps ("First come, first serve"). I'll let you know as soon as I pick it up again or need help. Is that ok for you?

**Comment by manekinekko** [comment]
No worries at all – that's how open source works!
